core/src/main/kotlin/xtdb/arrow/DecimalVector.kt

```diff
@@ -0,0 +1,52 @@
+package xtdb.arrow
+
+import org.apache.arrow.memory.BufferAllocator
+import org.apache.arrow.vector.types.pojo.ArrowType
+import xtdb.api.query.IKeyFn
+import xtdb.arrow.metadata.MetadataFlavour
+import xtdb.util.Hasher
+import java.math.BigDecimal
+
+class DecimalVector private constructor(
+    override var name: String, override var nullable: Boolean, override var valueCount: Int,
+    override val validityBuffer: ExtensibleBuffer, override val dataBuffer: ExtensibleBuffer,
+    val precision: Int, val scale: Int, val bitWidth: Int,
```

Comment:
I'd probably have the constructor take ArrowType.Decimal, then extract these three in the class body - chances are your caller already has the ArrowType in hand